components/core/highlight/highlight.pipe.ts

```diff
@@ -31,23 +31,30 @@ function encodeEntities(value: string): string {
     .replace(/>/g, '&gt;');
 }
 
+export function hightlightTransform(
+  value: string,
+  highlightValue: string,
+  flags?: string,
+  klass?: string
+): string | null {
+  const UNIQUE_WRAPPERS: [string, string] = ['##==-open_tag-==##', '##==-close_tag-==##'];
+  if (!highlightValue) {
+    return value;
+  }
+  // Escapes regex keyword to interpret these characters literally
+  const searchValue = new RegExp(highlightValue.replace(/([.*+?^=!:${}()|[\]\/\\])/g, '\\$&'), flags);
+  const wrapValue = value.replace(searchValue, `${UNIQUE_WRAPPERS[0]}$&${UNIQUE_WRAPPERS[1]}`);
+  return encodeEntities(wrapValue)
+    .replace(new RegExp(UNIQUE_WRAPPERS[0], 'g'), klass ? `<span class="${klass}">` : '<span>')
+    .replace(new RegExp(UNIQUE_WRAPPERS[1], 'g'), '</span>');
+}
+
```

Comment:
In order to use the same func as `highlightPipe` as the default highlight func, i have to export the func. Is there a better way?

Reply:
You can use this pipe just as a class:
```ts
const highlightPipe = new NzHighlightPipe();
highlightPipe.transform(...);
```